app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -0,0 +1,190 @@
+
+        results_that_cannot_pass_aamva.blank?
+      end
+
+      def threatmetrix_disabled_result
+        Proofing::DdpResult.new(
+          success: true,
+          client: 'tmx_disabled',
+          review_status: 'pass',
+        )
+      end
+
+      def out_of_aamva_jurisdiction_result
+        Proofing::StateIdResult.new(
+          errors: {},
+          exception: nil,
+          success: true,
+          vendor_name: 'UnsupportedJurisdiction',
+        )
+      end
+
+      def lexisnexis_ddp_proofer
+        @lexisnexis_ddp_proofer ||=
+          if IdentityConfig.store.lexisnexis_threatmetrix_mock_enabled
+            Proofing::Mock::DdpMockClient.new
+          else
+            Proofing::LexisNexis::Ddp::Proofer.new(
+              api_key: IdentityConfig.store.lexisnexis_threatmetrix_api_key,
+              org_id: IdentityConfig.store.lexisnexis_threatmetrix_org_id,
+              base_url: IdentityConfig.store.lexisnexis_threatmetrix_base_url,
+            )
+          end
+      end
+
+      def resolution_proofer
+        @resolution_proofer ||=
+          if IdentityConfig.store.proofer_mock_fallback
+            Proofing::Mock::ResolutionMockClient.new
+          else
+            Proofing::LexisNexis::InstantVerify::Proofer.new(
+              instant_verify_workflow: IdentityConfig.store.lexisnexis_instant_verify_workflow,
+              account_id: IdentityConfig.store.lexisnexis_account_id,
+              base_url: IdentityConfig.store.lexisnexis_base_url,
+              username: IdentityConfig.store.lexisnexis_username,
+              password: IdentityConfig.store.lexisnexis_password,
+              request_mode: IdentityConfig.store.lexisnexis_request_mode,
+            )
+          end
+      end
+
+      def state_id_proofer
+        @state_id_proofer ||=
+          if IdentityConfig.store.proofer_mock_fallback
+            Proofing::Mock::StateIdMockClient.new
+          else
+            Proofing::Aamva::Proofer.new(
+              auth_request_timeout: IdentityConfig.store.aamva_auth_request_timeout,
+              auth_url: IdentityConfig.store.aamva_auth_url,
+              cert_enabled: IdentityConfig.store.aamva_cert_enabled,
+              private_key: IdentityConfig.store.aamva_private_key,
+              public_key: IdentityConfig.store.aamva_public_key,
+              verification_request_timeout: IdentityConfig.store.aamva_verification_request_timeout,
+              verification_url: IdentityConfig.store.aamva_verification_url,
+            )
+          end
+      end
+
+      # Make a copy of pii with the user's state ID address overwriting the address keys
+      def with_state_id_address(pii)
+        pii.transform_keys(SECONDARY_ID_ADDRESS_MAP)
+      end
+
+      SECONDARY_ID_ADDRESS_MAP = {
+        state_id_address1: :address1,
+        state_id_address2: :address2,
+        state_id_city: :city,
+        state_id_state: :state,
+        state_id_zipcode: :zipcode,
+      }.freeze
```

Comment:
Merging this will also require coordination with work on https://github.com/18F/identity-idp/pull/8235 due to the field renaming.